ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceIT.scala

```diff
@@ -0,0 +1,178 @@
+    val start = System.nanoTime()
+    reset(ledgerId).zip(Future.successful((System.nanoTime() - start).nanos))
+  }
+
+  private def submitAndWait(req: SubmitAndWaitRequest): Future[Empty] =
+    CommandServiceGrpc.stub(channel).submitAndWait(req)
+
+  private def activeContracts(ledgerId: String, f: TransactionFilter): Future[Set[CreatedEvent]] =
+    FiniteStreamObserver[GetActiveContractsResponse](
+      ActiveContractsServiceGrpc
+        .stub(channel)
+        .getActiveContracts(GetActiveContractsRequest(ledgerId, Some(f)), _))
+      .map(_.flatMap(_.activeContracts)(collection.breakOut))
+
+  private def submitAndExpectCompletions(ledgerId: String, commands: Int): Future[Unit] =
+    for {
+      _ <- Future.sequence(
+        Vector.fill(commands)(
+          CommandSubmissionServiceGrpc
+            .stub(channel)
+            .submit(dummyCommands(LedgerId(ledgerId), UUID.randomUUID.toString))))
+      unit <- WaitForCompletionsObserver(commands)(
+        CommandCompletionServiceGrpc
+          .stub(channel)
+          .completionStream(
+            CompletionStreamRequest(
+              ledgerId = ledgerId,
+              applicationId = M.applicationId,
+              parties = Seq(M.party),
+              offset = Some(M.ledgerBegin)
+            ),
+            _))
+    } yield unit
+
+  "ResetService" when {
+
+    "state is reset" should {
+
+      "return a new ledger ID" in {
+        for {
+          lid1 <- getLedgerId()
+          lid2 <- reset(lid1)
+          throwable <- reset(lid1).failed
+        } yield {
+          lid1 should not equal lid2
+          IsStatusException(Status.Code.NOT_FOUND)(throwable)
+        }
+      }
+
+      "return new ledger ID - 20 resets" in {
+        Future
+          .sequence(Iterator.iterate(getLedgerId())(_.flatMap(reset)).take(20).toVector)
+          .map(ids => ids.distinct should have size 20L)
+      }
+
+      // 5 attempts with 5 transactions each seem to strike the right balance
+      // to complete before the 30 seconds test timeout in normal conditions while
+      // still causing the test to fail if something goes wrong
+      //
+      // the 10 seconds timeout built into the context's ledger reset will
+      // be hit if something goes horribly wrong, causing an exception to report
+      // waitForNewLedger: out of retries
+      "consistently complete within 5 seconds" in {
+        val numberOfCommands = 5
+        val numberOfAttempts = 5
+        Future
+          .sequence(
+            Iterator
+              .iterate(getLedgerId()) { ledgerIdF =>
+                for {
+                  ledgerId <- ledgerIdF
+                  _ <- submitAndExpectCompletions(ledgerId, numberOfCommands)
+                  (newLedgerId, timing) <- timedReset(ledgerId) if timing <= 5.seconds
+                } yield newLedgerId
+              }
+              .take(numberOfAttempts)
+          )
+          .map(_ => succeed)
```

Comment:
Even when the assertion is just a `succeed`, I think I'd still prefer a `for`, but I'm less fussed about this one.

Reply:
There are different opinions in the team (@gerolf-da usually prefers `map` when the call is very simple). If you don't have a strong argument against this one I'd keep it as it is right now, mostly out of sheer inertia. Ideally I would have wanted to use `all(measurements) should be <= threshold` but for some reason it doesn't compile.